src/main/java/org/broadinstitute/dsm/route/KitUploadRoute.java

```diff
@@ -252,61 +256,85 @@ else if (InstanceSettings.TYPE_NOTIFICATION.equals(behavior.getType())) {
+                String addressId = null;
+                if (address != null) {
+                    addressId = address.getId();
+                }
+                collaboratorSampleId = KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), shippingId,
+                        kitTypeId, kit.getParticipantId().trim(), collaboratorParticipantId,
+                        collaboratorSampleId, userId, addressId,
+                        errorMessage, externalOrderNumber, false);
+                kit.setShippingId(shippingId);
+                kit.setExternalOrderNumber(externalOrderNumber);
             }
-            if (collaboratorSampleId == null) {
-                errorMessage += "collaboratorSampleId was too long ";
+            catch (EasyPostException e) {
+                throw new RuntimeException("EasyPost addressId could not be received ", e);
             }
         }
-        try {
-            String shippingId = DDPKitRequest.UPLOADED_KIT_REQUEST + KitRequestShipping.createRandom(20);
-            Address address = easyPostUtil.getAddress(((KitUploadObject) kit).getEasyPostAddressId());
-            String addressId = null;
-            if (address != null) {
-                addressId = address.getId();
+        else {
+
+            if (kitRequestSettings.getCollaboratorSampleTypeOverwrite() != null) {
+                bspCollaboratorSampleType = kitRequestSettings.getCollaboratorSampleTypeOverwrite();
+            }
+            if (StringUtils.isNotBlank(collaboratorParticipantId)) {
+                collaboratorSampleId = KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                if (collaboratorParticipantId == null) {
+                    errorMessage += "collaboratorParticipantId was too long ";
+                }
+                if (collaboratorSampleId == null) {
+                    errorMessage += "collaboratorSampleId was too long ";
+                }
+            }
+            try {
+                String shippingId = DDPKitRequest.UPLOADED_KIT_REQUEST + KitRequestShipping.createRandom(20);
+                Address address = easyPostUtil.getAddress(((KitUploadObject) kit).getEasyPostAddressId());
+                String addressId = null;
+                if (address != null) {
+                    addressId = address.getId();
+                }
+                KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), shippingId,
+                        kitTypeId, kit.getParticipantId().trim(), collaboratorParticipantId,
+                        collaboratorSampleId, userId, addressId,
+                        errorMessage, kit.getExternalOrderNumber(), false);
+                kit.setShippingId(shippingId);
```

Comment:
same as 297-301 so you could move it after the if/else and use the same code for both